src/test/pmem2_badblock_mocks/pmem2_badblock_mocks.c

```diff
@@ -0,0 +1,375 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2019-2020, Intel Corporation */
+
+/*
+ * pmem2_badblock_mocks.c -- unit test for pmem2_badblock_*()
+ */
+
+#include "unittest.h"
+#include "out.h"
+#include "source.h"
+#include "badblocks.h"
+#include "pmem2_badblock_mocks.h"
+
+#define MAX_BAD_BLOCKS_NUMBER 10
+#define MAX_EXTENTS_NUMBER 8
+
+typedef struct badblock bad_blocks_array[MAX_BAD_BLOCKS_NUMBER];
+
+/* HW bad blocks expressed in 512b sectors */
+static bad_blocks_array hw_bad_blocks[] =
+{
+	/* fd == 99 - did not found any matching device */
+	{ {0, 0} },
+	/* fd % 50 == 1 - test #0 - no bad blocks */
+	{ {0, 0} },
+	/* fd % 50 == 2 - test #1 - 1 HW bad block */
+	{ {1, 1}, {0, 0} },
+	/* fd % 50 == 3 - test #2 - 6 HW bad blocks */
+	{ {4, 10}, {16, 10}, {28, 2}, {32, 4}, {40, 4}, {50, 2}, {0, 0} },
+	/* fd % 50 == 4 - test #3 - 7 HW bad blocks */
+	{ {2, 4}, {8, 2}, {12, 6}, {20, 2}, {24, 10}, {38, 4}, {46, 2}, \
+	{0, 0} },
+};
+
+/* file's bad blocks expressed in 512b sectors */
+static bad_blocks_array file_bad_blocks[] =
+{
+	/* fd == 99 - did not found any matching device */
+	{ {0, 0} },
+	/* fd % 50 == 1 - test #0 - no bad blocks */
+	{ {0, 0} },
+	/* fd % 50 == 2 - test #1 */
+	{ {0, 2}, {0, 0} },
+	/* fd % 50 == 3 - test #2 */
+	{ {4, 2}, {8, 2}, {12, 2}, {16, 2}, {20, 2}, {24, 2}, {28, 2}, \
+	{32, 2}, {40, 2}, {0, 0} },
+	/* fd % 50 == 4 - test #3 */
+	{ {4, 2}, {8, 2}, {12, 2}, {16, 2}, {20, 2}, {24, 2}, {28, 2}, \
+	{32, 2}, {40, 2}, {0, 0} },
+};
+
+/* file's extents expressed in 512b sectors */
+static struct extent files_extents[][MAX_EXTENTS_NUMBER] =
+{
+	/* fd == 99 - test_basic - did not found any matching device */
+	{ {0, 0, 0} },
+	/* fd % 50 == 1 - test #0 - no bad blocks */
+	{ {0, 0, 0} },
+	/* fd % 50 == 2 - test #1 */
+	{ {0, 0, 2}, {0, 0, 0} },
+	/* fd % 50 == 3 - test #2 */
+	{ {2, 2, 4}, {8, 8, 2}, {12, 12, 6}, {20, 20, 2}, {24, 24, 10}, \
+	{38, 38, 4}, {46, 46, 2}, {0, 0, 0} },
+	/* fd % 50 == 4 - test #3 */
+	{ {4, 4, 10}, {16, 16, 10}, {28, 28, 2}, {32, 32, 4}, {40, 40, 4}, \
+	{50, 50, 2}, {0, 0, 0} },
+};
+
+/*
+ * map_test_to_set -- map number of test to number of set of bad blocks
+ */
+static unsigned
+map_test_to_set(unsigned test)
+{
+	unsigned set = test % MASK_SET;
```

Comment:
this just seems odd to me, it would be far easier we could just do `test & TESTID_MASK`.

Reply:
Done